### PR TITLE
Geodashboard skeleton with text widget

### LINKED ISSUE #2592: Introduce text widget
### Description
Add a widget to add html text. 

### Acceptance Criteria - AC
 - The widget can be added both to the geodashboard and the map
 - User can set text with a WYSIWYG editor

### LINKED ISSUE #2593: Add geodashboard skeleton
### Description
Support base geodashboard editor and skeleton as for the [mockup] (https://demo.geo-solutions.it/mockups/mapstore2/#/dashboard). 

### Acceptance Criteria - AC
 - There should be a first version of editor for dashboard (reachable at `https://dev.mapstore2.geo-solutions.it/mapstore/#/dashboard) (not save yet) 
 - Charts and text widgets can be added to the dashboard

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
With this pull request we have a first skeleton of a geodashboard, to add features as for the [mockup](https://demo.geo-solutions.it/mockups/mapstore2/#/dashboard). 
The dashboard is accessible at #dashboard of the demo and supports charts and text widgets. 
## Issues
 - Fix #2592
 - Fix #2593

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Feature
**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [x] No

**Other information**:
This pull request contain many changes so a small guide can be useful for the review: 
**general**:
 - Introduced a `localizeProps` enhancer that allows to replace a msgId string with the localized string in props (for some components that need strings and not messages, like the input `placeholder`
 - Reorganized loaders a little bit to have the reusable tools as for the mockup. 
 - Introduced some enhancers for infinite scroll (for catalog inside widget builder)
**widgets builder**:
 - Enhancers and components that can be reused in any other future wizard of the current widget bulder (e.g. map, table, counter...)
 - Added some enhancer for the widget builder to select the widget type (and in case of chart the target layer)  before to start the wizard. This way the wizards can be reused
**catalog (for widget builder)**: 
 - Introduced a first reusable catalog that should replace the existing one in the future. This has been done by extracting some features from current catalog component into utility functions and using the above enhancers.
**page layout**: 
 - Added a `HolyGrail` plugin container that allow to render plugins in a borderlayout, so the plugins can be rendered into different areas via configuration. This is needed for the current dashboard and the future new home page
 - Added the text widget
 - Created the dashboard on top of all the changes above.

### CODE REVIEW

File: web/client/components/dashboard/Dashboard.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const emptyState = require('../misc/enhancers/emptyState');
+module.exports =
+    compose(
+        pure,
+        emptyState(
+            ({widgets = []} = {}) => widgets.length === 0,
+            () => ({
+                glyph: "dashboard",
+                title: "The dashboard is empty" // TODO i18n
+            })
```
Comment (MV88) [inline web/client/components/dashboard/Dashboard.jsx]: todo i18n

File: web/client/components/widgets/enhancers/textWidget.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright 2017, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+const {compose} = require('recompose');
+const deleteWidget = require('./deleteWidget');
+module.exports = compose(
```
Comment (mbarto) [inline web/client/components/widgets/enhancers/textWidget.js]: ???

File: web/client/containers/HolyGrail.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+const PluginsUtils = require('../utils/PluginsUtils');
+const ConfigUtils = require('../utils/ConfigUtils');
+const BorderLayout = require('../components/layout/BorderLayout');
+
+const PluginsContainer = connect((state) => ({
+    mode: urlQuery.mode || (urlQuery.mobile || state.browser && state.browser.mobile ? 'mobile' : 'desktop'),
+    monitoredState: PluginsUtils.getMonitoredState(state, ConfigUtils.getConfigProp('monitorState'))
+}))(require('../components/plugins/PluginsContainer'));
+
+class HolyGrail extends React.Component {
```
Comment (mbarto) [inline web/client/containers/HolyGrail.jsx]: HolyGrail?

**Comment by mbarto** [comment]
In general, it looks difficult to review such big PRs

File: web/client/containers/HolyGrail.jsx (hunk 3 above)
Comment (offtherailz) [reply web/client/containers/HolyGrail.jsx]: I mean this https://en.wikipedia.org/wiki/Holy_grail_(web_design)

File: web/client/components/widgets/enhancers/textWidget.js (hunk 2 above)
Comment (offtherailz) [reply web/client/components/widgets/enhancers/textWidget.js]: Yes, compose only one enhancer is unnecessary

**Comment by offtherailz** [comment]
Yes I know is a little hard, but I had to reorganize a lot of file to do that. Please take a look to the P.R. information to get the points you can focus on.

**Pull request merged**